pyfar/classes/coordinates.py

```diff
@@ -453,16 +711,57 @@ def set_cyl(self, points_1, points_2, points_3,
             unit in which the coordinate points are stored. The default is
             ``'rad'``.
         """
+        warnings.warn((
+            "This function will be deprecated in pyfar 0.8.0 in favor "
+            "of the cylindrical property."),
+                PyfarDeprecationWarning)
+        self._set_cyl(azimuth, z, radius_z, convention)
+
+    def _set_cyl(self, azimuth, z, rho, convention='top', unit='rad'):
+
+        # Convert to array
+        azimuth = np.asarray(azimuth)
+        z = np.asarray(z)
+        rho = np.asarray(rho)
+
+        # squeeze
+        if len(azimuth.shape) == 2 and \
+                (azimuth.shape[0] == 1 or azimuth.shape[1] == 1):
+            azimuth = azimuth.flatten()
+        if len(z.shape) == 2 and \
+                (z.shape[0] == 1 or z.shape[1] == 1):
+            z = z.flatten()
+        if len(rho.shape) == 2 and \
+                (rho.shape[0] == 1 or rho.shape[1] == 1):
+            rho = rho.flatten()
+
+        # convert to radians
+        if unit == 'deg':
+            azimuth = azimuth / 180 * np.pi
+        elif not unit == 'rad':
+            raise ValueError(
+                f"unit for {unit} is not implemented.")
```

Comment:
This doesn't sound right. The variable should be domain or system, right?

Reply:
or simply `f"{unit} is not implemented."`. I would be fine with leaving this as is since it will be deprecated anyhow.